crates/shared/src/order_validation.rs

```diff
@@ -522,6 +527,14 @@ impl OrderValidating for OrderValidator {
             full_fee_amount,
             class,
         )?;
+
+        if order.metadata.class == OrderClass::Limit {
+            self.limit_order_quoter
+                .quote(&order)
+                .await
+                .map_err(ValidationError::Other)?;
```

Comment:
Does this actually work?
As I understood it the order has to exist in the DB for the quoter to actually update the `surplus_fee` but the order will only be created once this function returns.
Also how should the program behave if the order is allowed to be created but the quoting process fails? This would only become relevant in the follow up PR, I guess. If it's so important to quote the order immediately quoting and order creation probably can't be 2 separate steps.